### PR TITLE
Status indicator box

### PR DESCRIPTION
**Pull request by MorganLove**
**DESCRIPTION**

- Adding a status indicator box at the top of the view

Note: For now, it only handles the relevant statuses for our current flow, and is viewable by anyone who is managing the audit, since we don't have separate user authorization and multiple jurisdictions yet. But It will be easy to add onto it as we implement those features.

**New Tests**

- Snapshots and downloading reports from top button 

**Checklist**

*Function*

[] Code is fully functional and ready for review (or is tagged WIP)

[] No errors/warnings in the browser console ( ﾟヮﾟ)/

[] package.json is committed if I added or removed any dependencies

[] Tests added as needed to ensure code base still has 100% passing coverage

[] Tests added as needed to cover new functionality

*Style*

[] Passes linter without errors

[] Code is easily understandable and self-documenting (using naming conventions in keeping with the rest of the codebase)

[] Interfaces all have 'I' before their names (as in `IAudit`)

[] Comments are only included as needed to explain the 'why' instead of the 'what'

[] Common standards and coding practices are adhered to

[] DRY: Don't Repeat Yourself. Code is reasonably structured to avoid repetition

[] Code is intuitively structured following a Separation of Concerns

[] No class components; only functional components (using Hooks as needed)

*Workflow*

[] Do an interactive rebase or a rebase

### CODE REVIEW

File: arlo-client/src/components/AuditForms/StatusBox.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    setup: [
+      'Audit setup is not complete.',
+      'Audit setup is complete.',
+      undefined,
+    ],
+  }
+
+  if (complete) {
+    // all jurisdictions have completed the audit
+    return [STATUSES.audit[3], STATUSES.setup[2]]
```
Comment (benadida) [inline arlo-client/src/components/AuditForms/StatusBox.tsx]: using status indexes here is fragile and hard to follow, what if you add one? I'd recommend using objects, e.g. `STATUSES.audit.in_progress`.

File: arlo-client/src/components/AuditForms/StatusBox.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    complete
+  )
+
+  return (
+    <Wrapper>
+      <H4>{auditStatus}</H4>
+      <div className="details">{setupStatus && <p>{setupStatus}</p>}</div>
+      <div className="wrapper">
+        <FormButton
+          disabled={!complete}
```
Comment (benadida) [inline arlo-client/src/components/AuditForms/StatusBox.tsx]: should we allow the report to be downloaded even if the audit isn't done?

File: arlo-client/src/components/AuditForms/StatusBox.tsx (hunk 1 above)
Comment (eventualbuddha) [reply arlo-client/src/components/AuditForms/StatusBox.tsx]: Definitely agree with @benadida here.

Comment (eventualbuddha) [reply arlo-client/src/components/AuditForms/StatusBox.tsx]: Or just remove the indirection altogether:

```ts
const generateStatuses = (
  audit: IAudit,
  launched: boolean,
  started: boolean,
  complete: boolean
): [string, string | undefined] => {
  if (complete) {
    // all jurisdictions have completed the audit
    return ['The audit is complete.', undefined]
  }
  if (launched && started) {
    // rounds have started
    return [
      `Round ${audit.rounds.length} of the audit is in progress.`,
      undefined,
    ]
  }
  if (launched && !started) {
    // after setup before first round
    return ['The audit has not started.', 'Audit setup is complete.']
  }
  /* istanbul ignore next */
  return ['The audit has not launched.', 'Audit setup is not complete.']
}
```

Comment (MorganLove) [reply arlo-client/src/components/AuditForms/StatusBox.tsx]: @eventualbuddha and @benadida I agree. I initially set it up as the array structure to help me visualize the flow, but you are right that it's better to do it without in the final product.

File: arlo-client/src/components/AuditForms/StatusBox.tsx (hunk 2 above)
Comment (MorganLove) [reply arlo-client/src/components/AuditForms/StatusBox.tsx]: @mcchilders Thoughts?

Would the scripts for generating the report work at earlier stages?

**Comment by MorganLove** [comment]
@mcchilders I'm thinking of redoing this to incorporate in the new changes when I rebase it. Just checking on how high priority that is and where in this week's timeline it would be best to slot that.
